File: tests/errors/ty_errors/ty_errors.fs
module rec ty_errors.Program

let main _ =
  let deepUnificationError =
    match ((1, 2), 3) with
    | (1, "type error"), 3 -> ()

  // ERROR! Index to int.
  // (0).[0]

  // ERROR! Recursive type.
  let rec f x = x f

  () // ERROR! This should be an int.

File: tests/errors/ty_errors/ty_errors.c
#error ty_errors:1:1 Type mismatch: 'unit' <> 'int'. Occurred while unifying 'unit' to 'int'.
#error ty_errors:3:5 Undefined variable f
#error ty_errors:7:24 Type mismatch: 'string' <> 'int'. Occurred while unifying '((int * string) * int)' to '((int * int) * int)'.
#error ty_errors:13:17 Recursive type occurred while unifying '{?11}@ty_errors:13:13' to '{?13}@ty_errors:13:3 -> {?22}@ty_errors:13:17'.
